src/LOAuth.ts

```diff
@@ -111,16 +131,28 @@ class LOAuth {
     state = JSON.stringify(state);
 
     /**
-     * Skip over the encoding when there is a 'state' query parameter.  This is the
-     * scenario when we are loading from the redirected auth endpoint and the state
-     * is already encoded in the URL parameter.  The problem is that encoding/decoding
-     * is being handled on multiple layers so that this client loads after the auth
-     * redirect, client-oauth will parse the state from the query param as unencoded,
-     * stringified JSON and when we pass an encoded version in, there is a comparison
-     * failure which results in a second redirect.
+     * When the auth flow is initiated there will be no 'state' query parameter.
+     * Because it is stringified JSON it needs to be encoded before passing it to the
+     * auth provider. If the user has to go through a sign in page to be
```

Comment:
Something about the "seems to contain something close to an encoded" combined with "redirect contains the state value but it is doubly encoded" seems like something is either not understood or something is not implemented quite right. Why would something be double encoded?